app/Http/Controllers/TrainingController.php

```diff
@@ -112,6 +114,18 @@ public function store(Request $request)
                 }
             }
 
+            //add credit
+
+            $credit_selected = $request->input('credit');
+            if($credit_selected == "1") {
```

Comment:
Perfect on store!
On update() we have to check if the credit hast change and update (create or delete) it if the change cause that

Reply:
public function store(Request $request)
    {
        if(Auth::user()->isAdmin()  || Auth::user()->isTrainingEditor()) {
            $training = new Training();
            $training->date = Carbon::createFromFormat('d m Y H:i', $request->get('date'));
            $training->dateEnd = empty($request->get('dateEnd')) ? null : Carbon::createFromFormat('d m Y H:i', $request->get('dateEnd'));
            $training->title = $request->get('title');
            $training->content = empty($request->get('content')) ? "" : $request->get('content');
            $training->sendbydatetime = empty($request->get('sendbydatetime')) ? null : Carbon::createFromFormat('d m Y H:i', $request->get('sendbydatetime'));
            $training->location = $request->get('location');
            $training->client_id = Auth::user()->currentclient_id;
            $training->save();

            //add positions of training
            if ($request->has('qualification') && $request->get('qualification')) {
                $qualifications = $request->get('qualification');
                $position_comment = $request->get('position_comment');
                for ($i = 0; $i < count($qualifications); $i++ ) {

                    if (Qualification::where(['id' => $qualifications[$i], 'client_id' => Auth::user()->currentclient_id])->count() > 0) {
                        $position = new Position();
                        $position->training_id = $training->id;
                        $position->service_id = null;
                        $position->qualification_id = $qualifications[$i];
                        $position->requiredposition = false;
                        $position->user_id = null;
                        $position->comment = $position_comment[$i];
                        $position->save();

                        //add credit
                            $credit_selected = $request->input('credit');
                            if($credit_selected == "1") {
                                $credit = new Credit();
                                $credit->position_id = $position->id;
                                $credit->qualification_id = $position->qualification_id;
                                $credit->points = '1';
                                $credit->save();
                        }
                    }
                }
            }


            Session::flash('successmessage', ' Neue Fortbildung erfolgreich angelegt');
            return redirect(action('TrainingController@create'));
        }

        abort(402, "Nope.");
    }



the methods needs to be like this otherwise it only creates a credit for the last position

Reply:
public function update(Request $request, $id)
    {
        if(!Auth::user()->isAdmin() && !Auth::user()->isTrainingEditor()) {
            abort(402, "Nope.");
        }
        $training = Training::findOrFail($id);

        if(!Auth::user()->isAdminOfClient($training->client_id) && !Auth::user()->isTrainingEditorOfClient($training->client_id)) {
            abort(402, "Nope.");
        }

        $training->date = Carbon::createFromFormat('d m Y H:i', $request->get('date'));
        $training->dateEnd = empty($request->get('dateEnd')) ? null : Carbon::createFromFormat('d m Y H:i', $request->get('dateEnd'));
        $training->title = $request->get('title');
        $training->content = empty($request->get('content')) ? "" : $request->get('content');
        $training->sendbydatetime = empty($request->get('sendbydatetime')) ? null : Carbon::createFromFormat('d m Y H:i', $request->get('sendbydatetime'));
        $training->location = $request->get('location');
        $training->save();

        //set changes and delete position
        if ($request->has('qualification') && $request->get('qualification')) {
            $qualifications = $request->get('qualification');
            $position_comment = $request->get('position_comment');
            $positions = $request->get('position');

            //remove deleted positions (with candidatures)
            if($request->has('delete_position')) {
                foreach ($request->get('delete_position') as $delete_position) {
                    if ($delete_position >= 0) {
                        Position::where('id', $delete_position)->forceDelete();
                    }
                }
            }

            //If there are more qualifications than position: User has Add new Positions
            $poscount = empty($positions) ? 0 : count($positions);
            if ($poscount < count($qualifications))
            {
                for ($i = $poscount; $i < count($qualifications); $i++ )
                {
                    if (Qualification::where(['id' => $qualifications[$i], 'client_id' => Auth::user()->currentclient_id])->count() > 0) {
                        $position = new Position();
                        $position->training_id = $training->id;
                        $position->service_id = null;
                        $position->qualification_id = $qualifications[$i];
                        $position->requiredposition = false;
                        $position->user_id = null;
                        $position->comment = $position_comment[$i];
                        $position->save();
                    }
                }
            }

            //are there changed positions
            for ($i = 0; $i < $poscount; $i++ ) {
                $pos = Position::findOrFail($positions[$i]);

                //just security reasons -> no manipulation of foreign positions
                if ($pos->training_id == $id){
                    //changed comment?
                    //=>Update without inform
                    if (strcmp($pos->comment, $position_comment[$i]) != 0) {
                        $pos->comment = $position_comment[$i];
                    }

                    //changed qualification and not null?
                    if ($pos->qualification_id != $qualifications[$i] && !empty($qualifications[$i])
                        &&  Qualification::where(['id' => $qualifications[$i], 'client_id' => Auth::user()->currentclient_id])->count() > 0)
                    {
                        $pos->qualification_id = $qualifications[$i];
                    }

                    $pos->save();

                    //add credit
                    $credit_selected = $request->input('credit');
                    if($credit_selected == "1") {
                        if($pos->getCredit($pos->id) == null) {
                            $credit = new Credit();
                            $credit->position_id = $pos->id;
                            $credit->qualification_id = $pos->qualification_id;
                            $credit->points = '1';
                            $credit->save();
                        }
                    }
                }
            }
        }